api/genericresource/helpers.go

```diff
@@ -0,0 +1,48 @@
+package genericresource
+
+import (
+	"fmt"
+
+	"github.com/docker/swarmkit/api"
+)
+
+// NewError creates a new error for Third Party Resources
+func NewError(format string, args ...interface{}) error {
+	return fmt.Errorf("Could not parse GenericResource: "+format, args...)
+}
+
+// NewSet creates a set object
+func NewSet(keys ...string) *api.GenericResource_Set {
+	keyMap := make(map[string]*api.Set_Void)
+
+	for _, k := range keys {
+		keyMap[k] = nil
+	}
+
+	return &api.GenericResource_Set{Set: &api.Set{Val: keyMap}}
+}
+
+// NewInteger creates an Integer object
+func NewInteger(val uint64) *api.GenericResource_Integer {
+	return &api.GenericResource_Integer{Integer: &api.Integer{Val: val}}
+}
+
+// NewResources creates a Resources object
+func NewResources() *api.Resources {
+	return &api.Resources{
+		Generic: make(map[string]*api.GenericResource),
+	}
+}
+
+// Count returns either the integer value of the Integer resource
+// or the len of the set resource
+func Count(val *api.GenericResource) uint64 {
```

Comment:
Agreed it's a helper function for the test to mask the switch I'll move it into the test